webapp/fmprod/src/test/java/org/apache/oodt/cas/product/jaxrs_rest/RestAPITest.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.oodt.cas.product.jaxrs_rest;
+
+import org.apache.cxf.endpoint.Server;
+import org.apache.cxf.jaxrs.JAXRSServerFactoryBean;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.jaxrs.lifecycle.SingletonResourceProvider;
+import org.apache.oodt.cas.product.jaxrs.resources.ProductResource;
+import org.apache.oodt.cas.product.jaxrs.services.CasProductJaxrsService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class RestAPITest extends Assert {
+    private final static String ENDPOINT_ADDRESS = "http://localhost:8080/cas_product_war/jaxrs/product?productId=" +"";
+    private static Server server;
+
+    @BeforeClass
+    public static void initialize() throws Exception {
+        startServer();
+    }
+
+    private static void startServer() throws Exception {
+        JAXRSServerFactoryBean sf = new JAXRSServerFactoryBean();
+        sf.setResourceProvider(CasProductJaxrsService.class,
+        new SingletonResourceProvider(new CasProductJaxrsService(), true));
+        sf.setResourceClasses(CasProductJaxrsService.class);
+        sf.setAddress(ENDPOINT_ADDRESS);
+        server = sf.create();
+    }
+
+    @AfterClass
+    public static void destroy() {
+        server.stop();
+        server.destroy();
+    }
+
+    @Test
+    public void testGetProductResourceWithWebClient() {
+        WebClient client = WebClient.create(ENDPOINT_ADDRESS);
+        ProductResource productResource = client.get(ProductResource.class);
+        assertEquals(0, productResource.getProductId());
```

Comment:
A suggestion. I assume the `ProductResource` returned here should be empty since we are not ingesting any file prior to fetching that. Isn't it better to add a section to ingest a file first and get the corresponding product here and assert?

Reply:
@NGimhana I think we should ignore this for now as you suggested on the call last week?

Reply:
Yes..I think it is safe to ignore this test at the current moment.